### PR TITLE
Fix invalid documentation about eachindex

### PR DESCRIPTION
**Pull request by DanielArndt**
This code doesn't execute. I believe due to a change that made it in before the 0.4.0 release. (perhaps this one? https://github.com/JuliaLang/julia/pull/12090)

I believe this is what was intended.

### CODE REVIEW

**Comment by tkelman** [comment]
This needs to be adjusted in the jl docstring since this section of the RST is auto populated. That section should be a doctest, we would have caught it faster.

**Comment by danielarndt** [comment]
ok. I'll look into that this evening, I haven't dealt with the auto generated / doctests yet.

**Comment by tkelman** [comment]
https://github.com/JuliaLang/julia/blob/master/CONTRIBUTING.md#improving-documentation describes how the system currently works, but let us know if you have any questions or have any suggested clarifications

**Comment by danielarndt** [comment]
@tkelman When I run `julia doc/genstdlib.jl`, *lots* of change are made to the documentation, most of which seem the wrong direction (ie. making the documentation worse). I ran with a fresh copy from master, rebuilt the whole language, and ran the script again -- same thing.

Perhaps others have been editing the docs directly as well?

**Comment by tkelman** [comment]
What version of julia is on your path? Make sure to run it with the just-built Julia, via `./julia doc/genstdlib.jl`

**Comment by danielarndt** [comment]
Ahh. Silly me. that fixed most of them.. Now there are still a couple changes, they seem to be spacing.

**Comment by yuyichao** [comment]
Yeah, there's a few space changes that doesn't seem to be synced in the rst doc. Feel free to include those in your PR.

**Comment by danielarndt** [comment]
OK. I think I did this right. Where can I read about doctests?

**Comment by yuyichao** [comment]
https://github.com/JuliaLang/julia/pull/13045

**Comment by danielarndt** [comment]
Reading through the comments in #13045, it seems to imply that non-deterministic output is going to cause problems with doctest (https://github.com/JuliaLang/julia/pull/13045#discussion_r39120492). 

Since this code is working off a randomly generated array, would this be a bad candidate for a doctest?

**Comment by tkelman** [comment]
ah, good point. we can change the construction from `sprand` to something deterministic using `sparse`

**Comment by danielarndt** [comment]
OK. I've switched the example out for a deterministic one. I'm not sure exactly how to test the doctest, but I'm assuming that since the build passed, it worked? It looks fine in the html output.

**Comment by tkelman** [comment]
We don't currently run the doctests on Travis, so I suspect there are a few unrelated failures on master right now. You can run them by `make -C doc doctest`, and compare the output before vs after this change to check that you don't introduce any new regressions, but the total number of tests in this file should increase.

File: base/docs/helpdb.jl [hunk 1]
```diff
@@ -3320,29 +3320,31 @@ Creates an iterable object for visiting each index of an AbstractArray `A` in an
...
-    A[iter] = 0.0
-    (iter.I_1,iter.I_2) = (1,2)
-    A[iter] = 0.02302469881746183
-    (iter.I_1,iter.I_2) = (2,2)
-    A[iter] = 0.0
-    (iter.I_1,iter.I_2) = (1,3)
-    A[iter] = 0.4864987874354343
-    (iter.I_1,iter.I_2) = (2,3)
-    A[iter] = 0.8090413606455655
+
```
Comment (tkelman) [inline base/docs/helpdb.jl]: the doctest might be picky about this newline, not positive

Comment (danielarndt) [inline base/docs/helpdb.jl]: It was! And I never would have noticed that that's the reason it "expected no output", so thanks for the pointer.

**Comment by danielarndt** [comment]
Ok, before:

```
Doctest summary
===============
 1012 tests
   64 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

And after:

```
Doctest summary
===============
 1014 tests
   64 failures in tests
    0 failures in setup code
    0 failures in cleanup code
```

File: base/docs/helpdb.jl [hunk 2]
```diff
@@ -3320,29 +3320,30 @@ Creates an iterable object for visiting each index of an AbstractArray `A` in an
...
 Example for a sparse 2-d array:
 
-    julia> A = sprand(2, 3, 0.5)
-    2x3 sparse matrix with 4 Float64 entries:
-        [1, 1]  =  0.598888
-        [1, 2]  =  0.0230247
-        [1, 3]  =  0.486499
-        [2, 3]  =  0.809041
+```jldoctest
+    julia> A = sparse([1, 1, 2], [1, 3, 1], [1, 2, -5])
```
Comment (tkelman) [inline base/docs/helpdb.jl]: from the other `jldoctest` examples, it looks like it shouldn't have a leading indent on the input lines in the markdown version

Comment (danielarndt) [inline base/docs/helpdb.jl]: Fixed

**Comment by danielarndt** [comment]
Anything else?

**Comment by tkelman** [comment]
Nope, looks perfect now. Thanks so much!

**Pull request merged**